chef/chef.py

```diff
@@ -147,32 +147,45 @@ def build_dir(self):
 
 
     def resolve_parent(self):
+        debug('resolv_parent({})'.format(self.name_))
         if self.inherit_:
-            if not self.inherit_ in BuildConfiguration.ALL:
-                fatal_error('target "{}"\'s parent "{}" not found in targets-section'
-                            .format(self.name_, self.inherit_))
-            self.parent_ = BuildConfiguration.ALL[self.inherit_]
-            self.parent_.add_child(self)
+            for parent in self.inherit_:
+                if not parent in self.parent_:
+                    if not parent in BuildConfiguration.ALL:
+                        fatal_error('target "{}"\'s parent "{}" not found in targets-section'
+                                    .format(self.name_, parent))
+                    self.parent_.append(parent)
+                    BuildConfiguration.ALL[parent].add_child(self)
 
 
     def target(self):
         return self.target_
 
 
     def layers(self):
+        layers = []
```

Comment:
Using a set() instead of an array for layers removes some `if not in`-statements.

```suggestion
        layers = set()
```